src/test/java/MacroExecutionModuleImplementationTest.java

```diff
@@ -0,0 +1,154 @@
+        Assert.assertEquals(expected, actual);
+    }
+
+    // Get macro name from the message when the macro has not yet been used in the thread
+    @Test
+    public void getMacroNameFromMessage() {
+        emptyThreadMacroMap();
+        String actual = execution.getMacroName(MESSAGE_WITH_MACRO_NAME, THREAD1);
+        Assert.assertEquals(MACRO_NAME, actual);
+    }
+
+    // Get macro name from the message when the macro has already been used in the thread (ie macro name is not stated explicitly in the message)
+    @Test
+    public void getMacroNameFromThread() {
+        addToThreadMacroMap(THREAD1, MACRO_NAME);
+        String actual = execution.getMacroName(MESSAGE_WITHOUT_MACRO_NAME, THREAD1);
+        Assert.assertEquals(MACRO_NAME, actual);
+    }
+
+    // Test the help message
+    @Test
+    public void testHelpMessage() throws IOException, GeneralSecurityException {
+        ChatResponse chatResponse = execution.getReplyText("@MacroBot /help", THREAD1, ROOM1, USER_EMAIL, HELP_MESSAGE);
+        String actual = chatResponse.getReplyText();
+        Assert.assertEquals(HELP_MESSAGE, actual);
+    }
+
+    // Test an empty message
+    @Test
+    public void testEmptyMessage() throws IOException, GeneralSecurityException {
+        ChatResponse chatResponse = execution.getReplyText("@MacroBot", THREAD1, ROOM1, USER_EMAIL, HELP_MESSAGE);
+        String actual = chatResponse.getReplyText();
+        Assert.assertEquals(EMPTY_MESSAGE_RESPONSE, actual);
+    }
+
+    // Test using a macro for the first time when you are the creator. In this case, scaulfeild@google.com created DailyBot
+    @Test
+    public void useYourMacroInNewRoom() throws IOException, GeneralSecurityException, Exception {
+        
+        //when(mockEntityModule.getMacro("scaulfeild@google.com", "DailyBot")).thenReturn(Optional.of(DEFAULT_MACRO));
+    
+        ChatResponse chatResponse = execution.getReplyText("@MacroBot DailyBot please log my message", THREAD1, ROOM1, "scaulfeild@google.com", HELP_MESSAGE);
+        String actual = chatResponse.getReplyText();
+        Assert.assertEquals("successfully executed", actual);
+    }
+
+/*
+    // Test using a macro for the first time when you are not the creator and it hasn't been shared (should be denied access)
+    @Test
+    public void usingAPrivateMacro() throws IOException, GeneralSecurityException {
+        ChatResponse chatResponse = execution.getReplyText("@MacroBot CongraBot hi", "spaces/AAAAAAAAAAA/threads/BBBBBBBBBBB", "spaces/AAAAAAAAAAA", "scaulfeild@google.com", HELP_MESSAGE);
+        String actual = chatResponse.getReplyText();
+        String macroName = execution.getMacroName("@MacroBot CongraBot hi", "spaces/AAAAAAAAAAA/threads/BBBBBBBBBBB");
+        String expected = (new ChatResponse(String.format("You do not own/have access to %s.", macroName))).getReplyText();
+
+        Assert.assertEquals(expected, actual);
+    }*/
+
+    // Test using someone else's macro after it has been shared
```

Comment:
Yeah that's the pattern that we should follow instead of testing the maps directly.